zaas-client/src/main/java/org/zowe/apiml/zaasclient/util/SimpleHttpResponse.java

```diff
@@ -0,0 +1,79 @@
+package org.zowe.apiml.zaasclient.util;
+
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpStatus;
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Simple record to hold data extracted from {@link org.apache.hc.core5.http.ClassicHttpResponse} so the underlying
+ * resources can be closed and the data processed further.
+ * Intended to be used with {@link org.apache.http.impl.client.CloseableHttpClient} execute methods as return
+ * value of {@link org.apache.http.client.ResponseHandler} in case of custom exceptions needs to be thrown during response processing.
+ *
+ * @param code       http status code
+ * @param byteBody   response body as byte[]
+ * @param stringBody response body as String
+ * @param headers    http response headers
+ */
+
+public record SimpleHttpResponse(
```

Comment:
What is a difference with `org.apache.hc.client5.http.async.methods.SimpleHttpResponse`?

Reply:
This one could be considered as alternative. However, it is in async package which could be confusing as there is no async communication. The new class is lightweight and also has a custom constructor that is used multiple times and we would need to share anyway.